lib/index.js

```diff
@@ -262,6 +262,23 @@ class Study extends EventTarget {
         die();
       }).bind(this)
     )
+    this.on(
+      "uninstall-custom-reason", (function (reason) {
+        prefSvc.get('shield.debug') && console.log("uninstall-custom-reason", this.states);
+        this.states.push("uninstall-custom-reason");
+        if (this.flags.dying) {
+          this.final();
+          return;
+        }
+        this.flags.dying = true;
+        this.report(merge({}, telemetrySubset(this.config), {study_state: reason}), "shield");
+        let that = this;
```

Comment:
I think you're right that it's not necessary to declare and use `that`. This also appears in lines 258 and 242, and I matched the structure there. I'll remove it in all 3 of those places.